fastapi/dev.nix

```
@@ -0,0 +1,28 @@
+{ pkgs, ... }: {
+  # Which nixpkgs channel to use.
+  channel = "stable-23.11"; # or "unstable"
+  # Use https://search.nixos.org/packages to find packages
+  packages = [
+    pkgs.python3
+    pkgs.python311Packages.pip
+    pkgs.python311Packages.fastapi
+    pkgs.python311Packages.uvicorn
+  ];
+  # Sets environment variables in the workspace
+  env = {};
+  idx = {
+    # Search for the extensions you want on https://open-vsx.org/ and use "publisher.id"
+    extensions = [ "ms-python.python" "rangav.vscode-thunder-client" ];
+    workspace = {
+      # Runs when a workspace is first created with this `dev.nix` file
+      onCreate = {
+        install =
+          "python -m venv .venv && source .venv/bin/activate &&  pip install -r requirements.txt";
+        # Open editors for the following files by default, if they exist:
+        default.openFiles = [ "app.py" ];
+      };
+      # To run something each time the workspace is (re)started, use the `onStart` hook
+      onStart = { run-server = "./devserver.sh"; };
```

Instead of `onStart` let's use the preview server instead.

```suggestion
      previews = {
      enable = true;
      previews = {
        web = {
          command = [ "./devserver.sh" ];
          env = { PORT = "$PORT"; };
          manager = "web";
        };
      };
    };
  };
```

---

We'll need to double check the formatting on this one. Multiline suggestions can be tricky.